### PR TITLE
naacl2013 archive video ingestion

### PR DESCRIPTION
**Pull request by xinru1414**

### CODE REVIEW

**Comment by mjpost** [comment]
Thanks for this! It looks good to me. Can you put example usage at the top of the script? It seems that it expects a directory that looks like this?

```
videos/
    N13-1001.m4v
    N13-1002.m4v
    ...
```

is that correct? What is the invocation, then?

**Comment by mjpost** [comment]
This doesn't seem quite right in the preview:

* It mixes up the button with the plain-text link. The link here should be just plain-text

image

* There should be a video button in the upper right
* This seems to have broken the video button in the listing files. See for example [here](https://aclanthology.org/volumes/D18-1/) (where it works) and [on the preview](https://preview.aclanthology.org/naacl-2013-video-ingestion/volumes/D18-1/) (where it doesn't work)

image

File: bin/anthology/papers.py
Comment (mjpost) [inline bin/anthology/papers.py:92]: I think in `master` branch videos must be handled as general attachments, but I don't remember offhand. This is a good direction, though; you just have to figure out why it broke the current videos.

**Comment by mbollmann** [comment]
What exactly are the changes in `bin/anthology/*.py` trying to achieve?

**Comment by mjpost** [comment]
The goal is to represent videos that are hosted locally. We'll need to do a little work here though since this isn't quite right. 

Treating videos as separately named from attachments might be a good idea but the decision is separate from the changes here.

File: bin/anthology/utils.py
Comment (mbollmann) [inline bin/anthology/utils.py:444]: This doesn't look right; value is a dict here for a reason.

**Comment by xinru1414** [comment]
tested on localhost: 
the video icons on the proceedings page
Screen Shot 2022-01-13 at 8 01 03 PM
the video icon on the paper page
Screen Shot 2022-01-13 at 8 00 54 PM

**Comment by mjpost** [comment]
Fantastic! They’re showing up in the preview, too. I’ll do a thorough review soon.

Looking through the [N13-1 preview](https://preview.aclanthology.org/naacl-2013-video-ingestion/volumes/N13-1), there is a chunk of missing videos (24–70). Any idea what happened here?

**Comment by xinru1414** [comment]
@mjpost videos (24–70) don't exist in the archive

File: bin/anthology/utils.py
Comment (mjpost) [inline bin/anthology/utils.py:444]: I'm guessing we don't use these other values here?

**Review by mjpost — APPROVED** [review]
LGTM. I like how you've split out videos as a special kind of attachment.

One question that comes to mind, though, is whether this handles multiple video attachments? As we continue ingesting these, we will have situations where we have both our local mirror of a video and a remotely-hosted version (say on slideslive). Perhaps we can handle that situation when we come to it.

It's a shame that we're missing so many videos from NAACL 2013...

**Comment by mjpost** [comment]
Let's wait till Monday to merge this in case @mbollmann or anyone else wants to comment.

**Comment by xinru1414** [comment]
@mjpost SG

File: bin/anthology/utils.py
Comment (mbollmann) [inline bin/anthology/utils.py:439]: This comment is now no longer true, should probably remove it

File: bin/anthology/papers.py
Comment (mbollmann) [inline bin/anthology/papers.py:18]: ? I don't see this used

File: bin/anthology/utils.py
Comment (mbollmann) [reply bin/anthology/utils.py:444]: These are the expected values for type "attachment", but "video" is now separate. I wonder if we might have a use for additional fields in the future, though, but it probably wouldn't be too difficult to refactor again later.

File: data/xml/Q13.xml
Comment (mbollmann) [inline data/xml/Q13.xml:38]: I'm not sure why we keep this `tag="video"` around. I've done a grep on all our current XML files and the value for `tag` is never something else than `"video"`. Since we also don't do anything with it in the code, I think we should just get rid of it now. @mjpost?

For reference:
```
$ grep "<video" *.xml | sed 's/.*tag="\([[:alpha:]]*\)".*/\1/' | sort | uniq -c                                                                                                                                 
   4010 video
```

File: hugo/layouts/papers/list-entry-author.html
Comment (mbollmann) [inline hugo/layouts/papers/list-entry-author.html:25]: Add the dashes here too for consistency (`{{- ... -}}`)

File: hugo/layouts/papers/list-entry.html
Comment (mbollmann) [inline hugo/layouts/papers/list-entry.html:26]: Add the dashes here too for consistency (`{{- ... -}}`)

File: data/xml/Q13.xml
Comment (mjpost) [reply data/xml/Q13.xml:38]: Yes, @xinru1414, let’s get rid of `tag=“video”`. We’ve had this discussion elsewhere before.

I also agree it might be good to use a dictionary for video values. For example, we might want to have a field that distinguishes remote from local links. We don’t need it now so we could just refactor later, but it might be easier now to preserve the dictionary structure while it’s fresh in your memory.

File: bin/anthology/utils.py
Comment (xinru1414) [reply bin/anthology/utils.py:444]: @mjpost @mbollmann Yes, it makes more sense to refactor in the future (with little effort).

File: data/xml/Q13.xml
Comment (akoehn) [reply data/xml/Q13.xml:38]: When getting rid of the tag, remember to update the schema and remove tags from the video element here: https://github.com/acl-org/acl-anthology/blob/3ef7ced577d9c95feaf2cecc94cd33d13398eea1/data/xml/schema.rnc#L72

Comment (xinru1414) [reply data/xml/Q13.xml:38]: @akoehn yep SG thanks for the reminder

**Comment by xinru1414** [comment]
@mjpost @mbollmann @akoehn I believe I've addressed all comments. Specifically for removing `tag=“video”`, I've removed it from the schema, all python scripts and all xml files (If I missed anything plz lmk).

**Review by mbollmann — APPROVED** [review]
LGTM now. Very nice!

**Comment by nschneid** [comment]
Very glad to see these videos restored!

Do you want feedback on broken links? E.g. the videos at https://preview.aclanthology.org/naacl-2013-video-ingestion/volumes/N13-4/ are giving 404s.

**Comment by mjpost** [comment]
@nschneid good catch. These files are split into parts and the htaccess regex didn't catch it. I just made an update attempt. This should work now once the preview is pushed out.

**Comment by xinru1414** [comment]
@mjpost thanks for the regex fix!

**Comment by mjpost** [comment]
It still wasn't quite working, but this latest update should do it.

**Comment by xinru1414** [comment]
@mjpost i think it's working now. i spot checked several links: https://preview.aclanthology.org/naacl-2013-video-ingestion/N13-4004.3.mp4 , https://preview.aclanthology.org/naacl-2013-video-ingestion/N13-4001.1.mp4 , https://preview.aclanthology.org/naacl-2013-video-ingestion/N13-4006.1.mp4

Shall I merge?

Also, after merging I think we can move on to archiving/processing more videos?

**Comment by mjpost** [comment]
Sure—please make sure to trim the commit message to just a summary of the relevant commits.

And yes, let’s move on to other conferences. Can you make a spreadsheet to track the status? I suggest (conference x year).

**Pull request merged**
